src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Semantic/Services/SemanticTokensEditsDiffer.cs

```diff
@@ -67,37 +67,59 @@ private Container<SemanticTokensEdit> ProcessEdits(IReadOnlyList<DiffEdit> diffs
                         }
                         else
                         {
-                            results.Add(new SemanticTokensEdit
+                            razorResults.Add(new RazorSemanticTokensEdit
                             {
                                 Start = diff.Position,
-                                Data = ImmutableArray<int>.Empty,
                                 DeleteCount = 1,
                             });
                         }
                         break;
                     case DiffEdit.Type.Insert:
                         if (current != null &&
-                            current.Data.HasValue &&
-                            current.Data.Value.Any() &&
+                            current.Data != null &&
+                            current.Data.Any() &&
                             current.Start == diff.Position)
                         {
-                            current.Data = current.Data.Append(NewArray[diff.NewTextPosition!.Value]).ToImmutableArray();
+                            current.Data.Add(NewArray[diff.NewTextPosition!.Value]);
                         }
                         else
                         {
-                            var semanticTokensEdit = new SemanticTokensEdit
+                            var semanticTokensEdit = new RazorSemanticTokensEdit
                             {
                                 Start = diff.Position,
-                                Data = ImmutableArray.Create(NewArray[diff.NewTextPosition!.Value]),
+                                Data = new List<int>{
```

Comment:
nit: format

Reply:
Interestingly Ctrl-K Ctrl-D doesn't fix this one. Will do by hand.